backend/pkg/api/data_access/archiver.go

```diff
@@ -67,3 +72,75 @@ func (d *DataAccessService) GetValidatorDashboardsInfo(ctx context.Context) (map
 
 	return result, nil
 }
+
+func (d *DataAccessService) UpdateValidatorDashboardsArchiving(ctx context.Context, dashboards []t.ArchiverDashboardArchiveReason) error {
+	ds := goqu.Dialect("postgres").Update("users_val_dashboards")
+
+	cases := goqu.Case()
+	for _, dashboard := range dashboards {
+		cases = cases.When(goqu.I("id").Eq(dashboard.DashboardId), dashboard.ArchivedReason.ToString())
+	}
+
+	ds = ds.Set(goqu.Record{"is_archived": cases})
+
+	// Restrict the query to the ids we want to set
+	ids := make([]interface{}, len(dashboards))
+	for i, dashboard := range dashboards {
+		ids[i] = dashboard.DashboardId
+	}
+	ds = ds.Where(goqu.I("id").In(ids...))
+
+	query, args, err := ds.Prepared(true).ToSQL()
+	if err != nil {
+		return fmt.Errorf("error preparing query: %v", err)
+	}
+
+	_, err = d.writerDb.ExecContext(ctx, query, args...)
+	return err
+}
+
+func (d *DataAccessService) RemoveValidatorDashboards(ctx context.Context, dashboardIds []uint64) error {
+	tx, err := d.alloyWriter.BeginTxx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("error starting db transactions to remove validator dashboards: %w", err)
+	}
+	defer utils.Rollback(tx)
+
+	// Delete the dashboard
```

Comment:
This should be done within a single `cascade` delete (foreign keys likely got lost in a migration though, need to fix that first)